core/frontend/src/main/scala/io/udash/properties/seq/TransformedSeqProperty.scala

```diff
@@ -1,46 +1,50 @@
 package io.udash.properties.seq
 
 import io.udash.properties.single.{Property, ReadableProperty}
-import io.udash.utils.Registration
 
 private[properties]
-class TransformedReadableSeqProperty[A, B, +ElemType <: ReadableProperty[B], OrigType <: ReadableProperty[A]]
+class TransformedReadableSeqProperty[A, B, ElemType <: ReadableProperty[B], OrigType <: ReadableProperty[A]]
                                     (override protected val origin: ReadableSeqProperty[A, OrigType], transformer: A => B)
-  extends ForwarderReadableSeqProperty[B, ElemType] {
-
-  protected def transformElement(el: OrigType): ElemType =
-    el.transform(transformer).asInstanceOf[ElemType]
-
-  override def get: Seq[B] =
-    origin.get.map(transformer)
-
-  override def elemProperties: Seq[ElemType] =
-    origin.elemProperties.map(p => transformElement(p))
-
-  override def listenStructure(structureListener: (Patch[ElemType]) => Any): Registration =
-    origin.listenStructure(patch =>
-      structureListener(Patch[ElemType](
-        patch.idx,
-        patch.removed.map(p => transformElement(p)),
-        patch.added.map(p => transformElement(p)),
-        patch.clearsProperty
-      ))
+  extends ForwarderReadableSeqProperty[A, B, ElemType, OrigType] with ForwarderWithLocalCopy[A, B, ElemType, OrigType] {
+
+  private var lastValue: Seq[A] = _
+  private var transformedLastValue: Seq[B] = _
```

Comment:
Is this significantly (or at all) faster than an immutable reference + mutable Seq?

Reply:
No, but I like this implementation (var + immutable Seq), because it always replaces the whole Seq.